pytorch_lightning/utilities/memory.py

```diff
@@ -87,3 +91,54 @@ def garbage_collection_cuda():
         if not is_oom_error(exception):
             # Only handle OOM errors
             raise
+
+
+def get_memory_profile(mode: str) -> Union[Dict[str, int], Dict[int, int]]:
+    """Get a profile of the current memory usage.
+
+    Args:
+        mode: There are two modes:
+
+            - 'all' means return memory for all gpus
+            - 'min_max' means return memory for max and min
+
+    Return:
+        A dictionary in which the keys are device ids as integers and
+        values are memory usage as integers in MB.
+        If mode is 'min_max', the dictionary will also contain two additional keys:
+
+        - 'min_gpu_mem': the minimum memory usage in MB
+        - 'max_gpu_mem': the maximum memory usage in MB
+    """
+    memory_map = get_gpu_memory_map()
+
+    if mode == "min_max":
+        min_index, min_memory = min(memory_map.items(), key=lambda item: item[1])
+        max_index, max_memory = max(memory_map.items(), key=lambda item: item[1])
+
+        memory_map = {"min_gpu_mem": min_memory, "max_gpu_mem": max_memory}
+
+    return memory_map
+
+
+def get_gpu_memory_map() -> Dict[str, int]:
```

Comment:
what does the `_map` stand for?

Reply:
@Borda these changes are mostly from the file rename. Imo we should make changes to improve the model summary in follow up PRs to minimize the number of changes here

Reply:
but it is map as a word or some shortcut?

Reply:
My guess is map as a word to indicate the return value is a Dict